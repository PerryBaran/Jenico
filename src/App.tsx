import React, {useState} from 'react';
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import songInfo from './songInfo';

import Navbar from './containers/Navbar';
import Home from './routes/Home';
import Music from './routes/Music';
import Album from './routes/Album';
import Contact from './routes/Contact';
import Player from './containers/Player';


function App() {
  const [data] = useState(songInfo);

  return (
    <Router>
      <Navbar/>
      <Routes>
        <Route path="/" element={<Home/>}/>
        <Route path="/music" element={<Music data={data}/>}/>
        <Route path="/music/:id" element={<Album/>}/>
        <Route path="/contact" element={<Contact/>}/>
      </Routes>
      <Player/>
    </Router>
  );
}

export default App;
