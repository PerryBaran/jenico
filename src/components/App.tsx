import { useState } from "react";
import { HashRouter as Router, Routes, Route } from "react-router-dom";
import songInfo from "../media/songInfo";
import useFirebaseData from "../hooks/useFirebaseData";
import Navbar from "./navbar/Navbar";
import Home from "./routes/home/Home";
import Music from "./routes/music/Music";
import Album from "./routes/album/Album";
import Contact from "./routes/contact/Contact";
import Player from "./player/Player";

function App() {
  const [playing, setPlaying] = useState(false);
  const [selectedSong, setSelectedSong] = useState(songInfo[0].songs[0].name);
  const [data] = useFirebaseData(songInfo, selectedSong);
  const [formFocused, setFormFocused] = useState(false);

  const handleFormFocus = (focus: boolean) => {
    setFormFocused(focus);
  };

  return (
    <Router>
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/music" element={<Music data={data} />} />
        <Route
          path="/music/:album"
          element={
            <Album
              data={data}
              playing={playing}
              setPlaying={setPlaying}
              selectedSong={selectedSong}
              setSelectedSong={setSelectedSong}
            />
          }
        />
        <Route
          path="/contact"
          element={<Contact handleFormFocus={handleFormFocus} />}
        />
      </Routes>
      <Player
        data={data}
        playing={playing}
        setPlaying={setPlaying}
        selectedSong={selectedSong}
        setSelectedSong={setSelectedSong}
        formFocused={formFocused}
      />
    </Router>
  );
}

export default App;
