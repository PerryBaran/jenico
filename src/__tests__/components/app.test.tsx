import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from "../../components/App";

describe("App", () => {
  afterEach(cleanup);

  // prevents error "Not implemented: HTMLMediaElement.prototype.pause"
  jest
    .spyOn(window.HTMLMediaElement.prototype, "pause")
    .mockImplementation(jest.fn());

  test("snapshot", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Navbar links", () => {
    render(<App />);

    expect(screen.getByText(/spotify/i)).toBeInTheDocument();
    fireEvent.click(screen.getByText(/music/i));
    expect(screen.getByText(/lysergic/i)).toBeInTheDocument();
    expect(screen.queryByText(/spotify/i)).not.toBeInTheDocument();
    fireEvent.click(screen.getByText(/jenico/i));
    expect(screen.getByText(/spotify/i)).toBeInTheDocument();
    fireEvent.click(screen.getByText(/contact/i));
    expect(screen.getByText(/email/i)).toBeInTheDocument();
    expect(screen.queryByText(/spotify/i)).not.toBeInTheDocument();
    fireEvent.click(screen.getByText(/home/i));
    expect(screen.getByText(/spotify/i)).toBeInTheDocument();
  });

  test("Music links", () => {
    render(<App />);

    fireEvent.click(screen.getByText(/music/i));
    fireEvent.click(screen.getByText(/ethereal/i));
    expect(screen.getByText(/distressed/i)).toBeInTheDocument();
  });
});
