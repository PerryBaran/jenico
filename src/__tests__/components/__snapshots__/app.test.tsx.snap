// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <nav
    class="container"
  >
    <a
      class="link"
      href="#/"
    >
      Jenico
    </a>
    <div />
    <a
      class="link"
      href="#/"
    >
      Home
    </a>
    <div />
    <a
      class="link"
      href="#/music"
    >
      Music
    </a>
    <div />
    <a
      class="link"
      href="#/contact"
    >
      Contact
    </a>
  </nav>
  <header
    class="container"
  >
    <div>
      <h1>
        Jenico
      </h1>
      <div>
        <a
          class="social"
          href="https://open.spotify.com/artist/6XqSoJQMIWNKBuLzzmJwiz"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Spotify icon"
            src="Spotify.png"
          />
          <p>
            Spotify
          </p>
        </a>
        <a
          class="social"
          href="https://jenico.bandcamp.com/"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Bandcamp icon"
            src="BandCamp.png"
          />
          <p>
            Bandcamp
          </p>
        </a>
        <a
          class="social"
          href="https://soundcloud.com/jenico_flac"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Soundcloud icon"
            src="Soundcloud.png"
          />
          <p>
            Soundcloud
          </p>
        </a>
        <a
          class="social"
          href="https://www.youtube.com/channel/UCOHuR2EY8wKXsDcocxDvqOA"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="YouTube icon"
            src="Youtube.png"
          />
          <p>
            YouTube
          </p>
        </a>
        <a
          class="social"
          href="https://www.instagram.com/jenico.flac/"
          rel="noreferrer"
          target="_Blank"
        >
          <img
            alt="Instagram icon"
            src="Instagram.png"
          />
          <p>
            Instagram
          </p>
        </a>
      </div>
    </div>
  </header>
  <div
    class="container"
  >
    <video
      autoplay=""
      class="hide"
      data-testid="video"
      loop=""
      src="DreamingOfDetunedLove.mp4"
    />
  </div>
  <section
    class="container"
  >
    <audio
      data-testid="audio"
    />
    <div
      class="container"
    >
      <img
        alt="Where We End cover art"
        src="WhereWeEnd.png"
      />
      <div>
        <h3>
          Where We End
        </h3>
        <p>
          Where We End
        </p>
      </div>
    </div>
    <div
      class="center"
    >
      <input
        class="bar"
        data-testid="progress-bar"
        max="0"
        min="0"
        name="time"
        type="range"
        value="0"
      />
      <div
        class="belowBar"
      >
        <div
          class="container"
        >
          <p>
            0:00
          </p>
        </div>
        <div
          class="container"
        >
          <button>
            <img
              alt="skip backward"
              src="skip.png"
            />
          </button>
          <button>
            <img
              alt="play/pause"
              src="play.png"
            />
          </button>
          <button>
            <img
              alt="skip forward"
              src="skip.png"
            />
          </button>
        </div>
        <div
          class="container"
        >
          <p>
            0:00
          </p>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <input
        max="100"
        min="0"
        name="volume"
        type="range"
        value="50"
      />
      <button>
        <img
          alt="volume"
          src="medium-volume.png"
        />
      </button>
    </div>
    <div
      class="container"
    >
      <div
        class="tracklist"
      >
        <img
          alt="Where We End cover art"
          src="WhereWeEnd.png"
        />
        <div>
          <button>
            &lt;
          </button>
          <h4>
            Where We End
          </h4>
          <button>
            &gt;
          </button>
        </div>
        <ul>
          <li
            class="highlight"
          >
            <button>
              Where We End
            </button>
          </li>
        </ul>
      </div>
      <button>
        <img
          alt="tracklist"
          src="menu.png"
        />
      </button>
    </div>
  </section>
</DocumentFragment>
`;
